### PR TITLE
Add Helm coverage - base case

### PR DESCRIPTION
**Pull request by pjgg**
### Summary

Test plan: https://github.com/quarkus-qe/quarkus-test-plans/blob/main/QUARKUS-1325.md
This pull request covers the base cases described on the above test plan. Needs Quarkus test framework 1.2.0.Beta8 or greater. 

Please select the relevant options.
- [X] New scenario (non-breaking change which adds functionality)

### Checklist:
- [X] Methods and classes used in PR scenarios are meaningful
- [X] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

File: helm/helm-app-database/src/test/resources/test.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ts.app.log.enable=false
+ts.global.generated-service.enabled=false # generated-service should be set to False on Helm scenarios
```
Comment (michalvavrik) [inline helm/helm-app-database/src/test/resources/test.properties:2]: There isn't too much information in that comment... I suppose it should be set to `false` as you set it to `false`, but maybe you can give a hint why (or if it's so clear, remove comment; not clear to me till I read more code)

File: helm/helm-app-database/pom.xml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            <!-- Disable native build on this module -->
+            <!-- Helm is concerned just about image name, Native compilation is not relevant -->
+            <id>native</id>
+            <activation>
+                <property>
+                    <name>native</name>
+                </property>
+            </activation>
+            <properties>
+                <!-- To not build the module on Native -->
```
Comment (michalvavrik) [inline helm/helm-app-database/pom.xml:71]: This comment is not precise, build is not skipped on native, but fast-jar is created. Stork is using same comment and when I run `mvn clean install` I got
```bash
[INFO] --- quarkus-maven-plugin:999-SNAPSHOT:build (build) @ stork ---
[WARNING] [io.quarkus.stork.deployment.SmallRyeStorkProcessor] The application is using the Stork Kubernetes Service Discovery provider but does not depend on the `quarkus-kubernetes-client` extension. It is highly recommended to use the `io.quarkus:quarkus-kubernetes-client` extension with the Kubernetes service discovery. 
To add this extension:
 - with the quarkus CLI, run: `quarkus ext add io.quarkus:quarkus-kubernetes-client`
 - with Apache Maven, run: `./mvnw quarkus:add-extension -Dextensions="io.quarkus:quarkus-kubernetes-client"`
 - or just add the `io.quarkus:quarkus-kubernetes-client` dependency to the project
[WARNING] [io.quarkus.vertx.web.deployment.ReactiveRoutesProcessor] Route io.quarkus.ts.stork.PongResource#pong() can match the same request and has the same order [0] as:
	- io.quarkus.ts.stork.PongReplicaResource#pong()
[WARNING] [io.quarkus.vertx.web.deployment.ReactiveRoutesProcessor] You can use @Route#order() to ensure the routes are not executed in random order
[INFO] [io.quarkus.deployment.QuarkusAugmentor] Quarkus augmentation completed in 1277ms
```

If your intention is truly skip the build, I'd suggest to set also
```
<quarkus.build.skip>true</quarkus.build.skip>
```
which leads to
```
[INFO] --- quarkus-maven-plugin:999-SNAPSHOT:build (build) @ stork ---
[INFO] Skipping Quarkus build
```
Alternatively, you can adjust the comment  (as well as `<!-- Disable native build on this module -->` one) so I can clearly understand what's your intention.

Thank you

File: helm/helm-app-database/pom.xml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            <artifactId>quarkus-resteasy</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-container-image-docker</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkiverse.helm</groupId>
+            <artifactId>quarkus-helm</artifactId>
+            <version>0.0.6-SNAPSHOT</version>
```
Comment (michalvavrik) [inline helm/helm-app-database/pom.xml:29]: Why did you add `io.quarkiverse.helm:quarkus-helm` to properties `<quarkus-helm.version>0.0.5</quarkus-helm.version>` and than set it here in higher version? And is the snapshot a way to go?

Comment (michalvavrik) [reply helm/helm-app-database/pom.xml:29]: [Quarkus Helm is currently using 2.12.2.Final](https://github.com/quarkiverse/quarkus-helm/blob/main/pom.xml#L27), won't it ever cause issues if we use it with f.e. 999-SNAPSHOT?

**Review by michalvavrik — CHANGES_REQUESTED** [review]
Please mention this module in _README_ and address or reply to my comments. Thank you

File: helm/helm-app-database/pom.xml (hunk 2 above)
Comment (pjgg) [reply helm/helm-app-database/pom.xml]: Thanks @michalvavrik I wan't aware about this property `<quarkus.build.skip>true</quarkus.build.skip>` :heart:

File: helm/helm-app-database/src/test/resources/test.properties (hunk 1 above)
Comment (pjgg) [reply helm/helm-app-database/src/test/resources/test.properties]: Code comment updated!

File: helm/helm-app-database/pom.xml (hunk 3 above)
Comment (pjgg) [reply helm/helm-app-database/pom.xml]: Was a leftover version. The right version was set on the main pom

**Comment by pjgg** [comment]
Current Pipeline error: Camel - Upstream issue: https://github.com/quarkusio/quarkus/issues/28007
```
Error: [ERROR] 	[error]: Build step org.apache.camel.quarkus.core.deployment.InjectionPointsProcessor#injectedComponents threw an exception: java.lang.ClassCastException: class org.jboss.jandex.MethodParameterInfo cannot be cast to class org.jboss.jandex.Type (org.jboss.jandex.MethodParameterInfo and org.jboss.jandex.Type are in unnamed module of loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @62d04f9d)
```

File: README.md [hunk 4]
```diff
@@ -369,6 +369,11 @@ This module covers some basic scenarios around GraphQL.
 ### `http/graphql-telemetry`
 This module covers some basic scenarios around OpenTelemetry usage with GraphQL.
 
+### `helm/helm-minimum`
+This module covers Helm integration by Quarkus-helm extension`io.quarkiverse.helm:quarkus-helm`
+The above extension will generate a Helm chart based on your `application.properties`. We have set up
+a HelloWorld application and then deploy it into Openshift through Helm
```
Comment (michalvavrik) [inline README.md:375]: Dot.

Comment (pjgg) [reply README.md]: done!

**Comment by michalvavrik** [comment]
Thanks, it looks like build is failing over 
```
[ERROR] 	[error]: Build step io.quarkus.container.image.docker.deployment.DockerProcessor#dockerBuildFromJar threw an exception: java.lang.RuntimeException: Execution of 'docker push quay.io/quarkusqeteam/ts-quarkus-helm-simple-app:v2' failed. See docker output for more details
```
To be honest, same happens to me locally, I have incorrectly set docker credentials (?).

**Comment by mjurc** [comment]
Run tests

**Comment by pjgg** [comment]
Run tests

**Comment by michalvavrik** [comment]
run tests

**Comment by pjgg** [comment]
run tests

**Review by michalvavrik — APPROVED** [review]
LGTM

**Comment by michalvavrik** [comment]
@pjgg One last thing, please don't merge this till you resolve `OpenShiftHelmSimpleAppIT#deployHelmQuarkusApplication` in OC, thank you

**Comment by pjgg** [comment]
> @pjgg One last thing, please don't merge this till you resolve `OpenShiftHelmSimpleAppIT#deployHelmQuarkusApplication` and `OpenShiftHelmSimpleAppIT#customReadiness` in OC, thank you
> 
> It would be very nice to have this in our tests https://quarkiverse.github.io/quarkiverse-docs/quarkus-helm/dev/index.html#_mapping_multiple_properties_at_once as we don't have `quarkus.helm.values[0].property` and `quarkus.helm.values[0].paths` tested.

This is going to be covered in another scenario called `helm-advanced` that will cover HelmDepencencies. Currently is blocked by an issue related to fields mapping

**Comment by mjurc** [comment]
Run tests

**Comment by pjgg** [comment]
Run tests

**Comment by pjgg** [comment]
Only `OperatorOpenShiftAmqStreamsKafkaStreamIT` has failed and is not related to this PR
Helm [scenarios](https://main-jenkins-csb-quarkusqe.apps.ocp-c1.prod.psi.redhat.com/job/quarkus-main-rhel8-jdk17-openshift-ci-distributed-ts-jvm-ocp-4/4/jdk=openjdk-11,label=RHEL8%20&&%20large,scenario=root-modules/testReport/io.quarkus.ts.helm.minimum/) succeed!
